src/apps/eos/actions/__init__.py

```diff
@@ -0,0 +1,117 @@
+        writers.write_action_deleteauth(w, action.delete_auth)
+    elif name == "linkauth":
+        await layout.confirm_action_linkauth(ctx, action.link_auth)
+        writers.write_action_linkauth(w, action.link_auth)
+    elif name == "unlinkauth":
+        await layout.confirm_action_unlinkauth(ctx, action.unlink_auth)
+        writers.write_action_unlinkauth(w, action.unlink_auth)
+    elif name == "newaccount":
+        await layout.confirm_action_newaccount(ctx, action.new_account)
+        writers.write_action_newaccount(w, action.new_account)
+    elif name == "transfer":
+        await layout.confirm_action_transfer(ctx, action.transfer, account)
+        writers.write_action_transfer(w, action.transfer)
+    else:
+        await process_unknown_action(ctx, w, action)
+
+    writers.write_action_common(sha, action.common)
+    writers.write_variant32(sha, len(w))
+    writers.write_bytes(sha, w)
+
+
+async def process_unknown_action(ctx, w, action):
+    checksum = HashWriter(sha256())
+    writers.write_variant32(checksum, action.unknown.data_size)
+    checksum.extend(action.unknown.data_chunk)
+
+    writers.write_bytes(w, action.unknown.data_chunk)
+    bytes_left = action.unknown.data_size - len(action.unknown.data_chunk)
+
+    while bytes_left != 0:
+        action = await ctx.call(
+            EosTxActionRequest(data_size=bytes_left), EosTxActionAck
+        )
+
+        if action.unknown is None:
+            raise ValueError("Bad response. Unknown struct expected.")
+
+        checksum.extend(action.unknown.data_chunk)
+        writers.write_bytes(w, action.unknown.data_chunk)
+
+        bytes_left -= len(action.unknown.data_chunk)
+        if bytes_left < 0:
+            raise ValueError("Bad response. Buffer overflow.")
+
+    await layout.confirm_action_unknown(ctx, action.common, checksum.get_digest())
+
+
+def check_action(action, name, account):
+    if account == "eosio":
+        if (
+            (name == "buyram" and action.buy_ram is not None)
+            or (name == "buyrambytes" and action.buy_ram_bytes is not None)
+            or (name == "sellram" and action.sell_ram is not None)
+            or (name == "delegatebw" and action.delegate is not None)
+            or (name == "undelegatebw" and action.undelegate is not None)
+            or (name == "refund" and action.refund is not None)
+            or (name == "voteproducer" and action.vote_producer is not None)
+            or (name == "updateauth" and action.update_auth is not None)
+            or (name == "deleteauth" and action.delete_auth is not None)
+            or (name == "linkauth" and action.link_auth is not None)
+            or (name == "unlinkauth" and action.unlink_auth is not None)
+            or (name == "newaccount" and action.new_account is not None)
+        ):
+            return True
+        else:
+            return False
+
+    elif name == "transfer":
+        return True if action.transfer is not None else False
```

Comment:
instead: `return action.transfer is not None`

Reply:
Done